roles/openshift_hosted/tasks/registry.yml

```diff
@@ -49,17 +49,7 @@
     openshift_hosted_registry_namespace: "{{ openshift.hosted.registry.namespace | default('default') }}"
     openshift_hosted_registry_selector: "{{ openshift.hosted.registry.selector }}"
     openshift_hosted_registry_images: "{{ openshift.hosted.registry.registryurl | default('openshift3/ose-${component}:${version}')}}"
-    openshift_hosted_registry_storage_glusterfs_path: "{{ openshift_hosted_registry_storage_glusterfs_path | default('') }}"
-    openshift_hosted_registry_storage_glusterfs_ips: "{{ openshift_hosted_registry_storage_glusterfs_ips | default('') }}"
-
-- name: Check if we use external gluster as backend
-  set_fact:
-    openshift_hosted_registry_external_gluster: True
-  when:
-  - openshift.hosted.registry.storage.kind | default(none) in ['glusterfs']
-  - openshift_hosted_registry_storage_glusterfs_path != ''
-  - openshift_hosted_registry_storage_glusterfs_ips != ''
-  - not 'glusterfs' in groups | default([])
+    openshift_hosted_registry_storage_glusterfs_ips: "{% set comma = joiner(',') %}{% if groups.glusterfs_registry is defined %}{% for node in groups.glusterfs_registry %}{{ comma() }}{{  hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip)}}{% endfor %}{% elif groups.glusterfs is defined %}{% for node in groups.glusterfs %}{{ comma() }}{{ hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip)}}{% endfor %}{% else %}{{ openshift_hosted_registry_storage_glusterfs_ips }}{% endif %}"
```

Comment:
A few things here:
* The current order of precedence is `glusterfs_registry` -> `glusterfs` -> `glusterfs_ips`. Do you think it would make sense to move `glusterfs_ips` as the first choice?
* Is the 'joiner' method preferred to using `{% if not loop.first %}` or something? I don't have an opinion on that, I just haven't seen `joiner` before.
* Add a space after `default(hostvars[node].openshift.common.ip)` :)

Reply:
@jarrpa i think order is fine now. As it populate variables later.

For joiner - will do some tests. Had some issues with standart join due format comming from ansible.

Reply:
Tests fails as I missed one usecase. Will fix later

Reply:
I meant something like `loop.last` in jinja2: http://jinja.pocoo.org/docs/2.9/templates/#for

Reply:
If you want I can change to loop.last but Joiner looks more nicer for me :D as it requires less if else conditions.

Reply:
Nah, that's fine! I'd just never seen it before. :) I'm fine leaving it as-is.